### ISSUE TITLE
fromVar should return an Either String a

### ISSUE THREAD

**Issue by dmjio** [dmjio/envy#30]
For more parse failure information, we should use the latest and greatest `readEither` function from `Text.Read.readEither`.

**Comment by dmjio** [comment]
Also note for decoding basic type the failure information returned isn't that useful, so this change might not be worth it.

```>>> readEither "hello" :: Either String Int
Left "Prelude.read: no parse"```

**Comment by cideM** [comment]
I'm trying to use `URI` from `Text.URI` together with Envy but I don't know how to line up the types **and** preserve error information. I started by wrapping `URI` in a newtype wrapper `Url` so that I don't get orphan instance warnings. GHC then tells me that 

```
No instance for (Envy.Var Url)
```

which makes sense. So I implement `Var` but the `fromVar` method has the type `fromVar :: String -> Maybe a` so I'm immediately stuck because I need `String -> Either Whatever Url` or any type that let's me add a custom error message or error information.

Then I started wondering about the purpose of `fromEnv :: Maybe a -> Parser a` since `Parser a` clearly let's me return more error information. But that information is already lost when I'm trying to implement `fromVar`... :thinking:

So long story short: the fact that `fromVar` already throws out all error information is to me a complete show stopper when it comes to Envy. Which is a shame because it seems that Haskell doesn't have many nice libraries in the env var parsing space :disappointed:

**Comment by dmjio** [comment]
@cideM we can change `fromVar` to be `fromVar :: String -> Either String a`. I'd support that change.
